.github/workflows/ci-Instrumentation.Process.yml

```diff
@@ -0,0 +1,41 @@
+name: Build OpenTelemetry.Instrumentation.Process
+
+on:
+  pull_request:
+    branches: [ 'main*', 'instrumentation*' ]
```

Comment:
we dont have any instrument** branches right?

Reply:
This was copied from the ci.yml file, I don't know why it was there, guess it's related to some overall practice (e.g. folks can branch out the instrumentation work)?

Reply:
I just removed it so we start from the minimal thing.